src/lib/tuning/abstract_evaluator.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "tuning/tuning_choice.hpp"
+
+namespace opossum {
+
+/**
+ * An AbstractEvaluator analyzes the current system state and proposes a list
+ * of TuningChoices that might improve the system performance.
+ */
+class AbstractEvaluator {
+ public:
+  virtual ~AbstractEvaluator() {}
+  /**
+   * Generate TuningChoices and append them to the given vector.
+   */
+  virtual void evaluate(std::vector<std::shared_ptr<TuningChoice>>& choices) = 0;
```

Comment:
When would you pass in a non-empty list? I am trying to understand why evaluate does not return the vector

Reply:
There might be multiple Evaluators in one Tuning operation which would all contribute TuningChoices to the vector. That is why the vector is created by the Tuner and handed to each Evaluator.